src/svm/impls/binary/binary.c

```diff
@@ -1549,6 +1708,73 @@ PetscErrorCode SVMGridSearch_Binary(SVM svm)
   PetscFunctionReturn(0);
 }
 
+#undef __FUNCT__
+#define __FUNCT__ "SVMLoadMatGramian_Binary"
+PetscErrorCode SVMLoadMatGramian_Binary(SVM svm,PetscViewer v)
+{
+  MPI_Comm comm;
+  Mat      G;
+
+  PetscFunctionBegin;
+  TRY( PetscObjectGetComm((PetscObject) svm,&comm) );
+
+  /* Create matrix */
+  TRY( MatCreate(comm,&G) );
+  TRY( MatSetType(G,MATDENSE) );
+  TRY( PetscObjectSetName((PetscObject) G,"G") );
+  TRY( PetscObjectSetOptionsPrefix((PetscObject) G,"gramian_") );
```

Comment:
inherit prefix from the svm object?

Reply:
Basically, it's a good idea when we think about complex machine learning application containing thousands of components, e.g. classifiers, dictionaries, autoencoders, data analysis part etc. I think current development stage of the package is not ready for implementing in such complex application because some necessities and standard functionalities missing like multiclass/multilabel approaches, implicit/explicit kernels, one-class classification, regressions (logistic, isotonic), v-SVM, lp-SVM, etc. Moreover, applying this idea should also be done in other parts of package, e.g. loading training/test dataset. Therefore, I've decided to not implement it in this PR.